inc/core.php

```diff
@@ -55,3 +55,12 @@ function get_package_path($id, $version) {
 	// This is safe - These values have been validated via validateIdAndVersion above
 	return '/packagefiles/' . $id . '/' . $version . '.nupkg';
 }
+
+/* Used to construct URIs */
+function http_https() {
```

Comment:
I am far from an expert in PHP; I just asked [StackOverflow](https://stackoverflow.com/questions/7304182/detecting-ssl-with-php) about the most reliable way to detect HTTPS, all pointed to the [`$_SERVER` array](http://www.php.net/manual/en/reserved.variables.server.php)